main.go

```diff
@@ -38,10 +39,19 @@ func main() {
 		log.Fatalf("Error converting nostr privkey to pubkey: %v", err)
 	}
 
-	db, err := gorm.Open(postgres.Open(cfg.DatabaseUri), &gorm.Config{})
-	if err != nil {
-		log.Fatalf("Failed to open DB %v", err)
+	var db *gorm.DB
+	if cfg.DatabaseBackendType == "POSTGRESS" {
```

Comment:
you could do 

```go
if strings.HasPrefix(cfg.DatabaseUri, "postgres://") || strings.HasPrefix(cfg.DatabaseUri, "postgresql://") || strings.HasPrefix(cfg.DatabaseUri, "unix://") {
...

} else {
  db, err = gorm.Open(sqlite.Open(cfg.DatabaseUri), &gorm.Config{})
  ...
}
```
so it checks if the DATABASE_URI is a postgres connection string then it uses PG otherwise it defaults to sqlite. 
then no additional config would be needed and it's possible to also use custom path for the SQLite: 
https://github.com/mattn/go-sqlite3#connection-string

Reply:
Why not drop the `||` and just check for the prefix `postgres`?